homepage/homepage/content/docs/upgrade/0-20-0.mdx

```diff
@@ -0,0 +1,293 @@
+import { Alert, CodeGroup } from '@/components/forMdx'
+
+# Jazz 0.20.0 - Full native crypto
+
+With this release, we complete our migration to a pure **native** Rust toolchain.
+We have removed the JavaScript compatibility layer, meaning our native Rust core now runs everywhere: React Native, Edge runtimes, all server-side environments, and the web.
+
+## Motivation
+
+Before 0.20.0, Jazz relied on a JavaScript crypto implementation to work around compatibility issues in some runtimes-most notably React Native.
+As a result, our native Rust core wasn’t running in React Native and, on some Edge runtimes, 
+only worked when WASM was explicitly initialized (see [WASM on Edge runtimes](/docs/react/server-side/setup#wasm-on-edge-runtimes)).
+
+The JavaScript crypto implementation is much slower than native Rust crypto. Although workarounds like RNQuickCrypto for React Native improved performance, they still only wrapped certain native libraries, rather than running Jazz's full Rust crypto.
+
+With native Rust crypto now running everywhere, Jazz crypto is faster and more reliable. 
+This also helps us speed up the migration of Jazz Core to Rust which will improve the Jazz overall performance.
+
+## What changed
+
+- We dropped support for `PureJSCrypto`. Jazz crypto now runs natively in each environment.
+- We dropped support for `RNQuickCrypto`. RNCrypto is now the default crypto provider on React Native.
+- There is no more fallback to Javascript crypto. If the crypto fails to initialize, it will throw an error. 
+**NB**: if you use WASM crypto on Edge runtimes, you need to ensure that WASM is available and initialized. See more details at [WASM on Edge runtimes](/docs/react/server-side/setup#wasm-on-edge-runtimes).
+
+## Breaking changes
+
+### Edge runtimes
+
+If you use WASM crypto on Edge runtimes, you need to ensure that WASM is available and initialized, there is no more fallback to Javascript crypto. 
+See more details at [WASM on Edge runtimes](/docs/react/server-side/setup#wasm-on-edge-runtimes).
+
+<CodeGroup>
+```ts
+import "jazz-tools/load-edge-wasm";
+// Other Jazz Imports
+
+export default {
+  async fetch(request: Request, env: Env, ctx: ExecutionContext) {
+    // Jazz application logic
+    return new Response("Hello from Jazz on Cloudflare!");
+  },
+};
+```
+</CodeGroup>
+
+if you don't do this, you will get an error like this:
+
+<CodeGroup>
+```
+Error: Failed to initialize WasmCrypto. WebAssembly is required for crypto operations on web platforms.
```

Comment:
This error should be more informative, either by giving the instructions to fix or by giving the doc page.

The former is preferred so AIs are more likely to fix the issue automatically (thinking about v0)